test/e2e/s3-e2e-specs.js

```diff
@@ -0,0 +1,41 @@
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import request from 'request-promise';
+import path from 'path';
+import { tempDir, fs, zip } from 'appium-support';
+import { uploadZip, deleteZip } from '../../lib/s3';
+import sinon from 'sinon';
+
+const {openDir} = tempDir;
+const {readFile, writeFile, mkdir} = fs;
+
+chai.should();
+chai.use(chaiAsPromised);
+
+describe('S3', () => {
+  let tempDir, testDir;
+
+  before(async () => {
+    // Create a temporary directory with one file and upload it to S3
+    tempDir = await openDir('temp-e2e-test-dir');
+    testDir = path.resolve(tempDir, 'test-dir');
+    await mkdir(testDir);
+    const fileContents = 'Temporary file contents';
+    await writeFile(path.resolve(testDir, 'temp-file.txt'), fileContents);
+  });
+
+  it('should zip directories and publish them to S3 and then unpublish them', async () => {
+    // Don't run the tests if AWS credentials are not set
+    if (!process.env.AWS_ACCESS_KEY_ID || !process.env.AWS_SECRET_ACCESS_KEY) {
+      return;
```

Comment:
we want to actually `skip()`, not `return`, so it doesn't falsely show up as passed. which unfortunately means no using arrow functions because it messes up the context.